encodings/fastlanes/src/delta/compress.rs

```diff
@@ -57,25 +53,22 @@ where
     // Loop over all the 1024-element chunks.
     if num_chunks > 0 {
         let mut transposed: [T; 1024] = [T::default(); 1024];
-        let mut base = [T::default(); T::LANES];
 
         for i in 0..num_chunks {
             let start_elem = i * 1024;
             let chunk: &[T; 1024] = array_ref![array, start_elem, 1024];
             Transpose::transpose(chunk, &mut transposed);
 
             // Initialize and store the base vector for each chunk
-            // TODO(ngates): avoid copying the base vector
-            base.copy_from_slice(&transposed[0..T::LANES]);
-            bases.extend(base);
+            bases.extend(&transposed[0..T::LANES]);
 
             deltas.reserve(1024);
             let delta_len = deltas.len();
             unsafe {
                 deltas.set_len(delta_len + 1024);
                 Delta::delta(
                     &transposed,
-                    &base,
+                    &*(transposed[0..T::LANES].as_ptr() as *const [_; T::LANES]),
```

Comment:
this is what `array_ref` macro does but because it generates functions the `T::LANES` const is not accessible

Reply:
can this be done with a transmute?

Reply:
You cannot since the compiler things the types are of different widths